turbopack/crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -2556,6 +2622,49 @@ async fn resolve_import_map_result(
                 ))
             }
         }
+        ImportMapResult::External(name, ty, traced) => Some(
+            ResolveResult::primary(ResolveResultItem::External {
+                name: name.clone(),
+                ty: *ty,
+                traced: *traced,
+            })
+            .cell(),
+        ),
+        ImportMapResult::AliasExternal(name, ty, traced, alias_lookup_path) => {
+            let request = Request::parse_string(name.clone());
+
+            // We must avoid cycles during resolving
+            if request.resolve().await? == *original_request
+                && alias_lookup_path.to_resolved().await? == original_lookup_path
+            {
+                None
+            } else if !(resolve_internal(
+                **alias_lookup_path,
+                request,
+                match ty {
+                    ExternalType::Url => options,
+                    // TODO is that root correct?
+                    ExternalType::CommonJs => node_cjs_resolve_options(alias_lookup_path.root()),
+                    ExternalType::EcmaScriptModule => {
+                        node_esm_resolve_options(alias_lookup_path.root())
+                    }
+                },
+            )
+            .await?
+            .is_unresolvable_ref())
```

Comment:
Could we pull out this expression into a variable? That complex expression inside of the `if` condition is a bit unreadable